### ISSUE TITLE
Allow plugins test run before running extractors

### ISSUE THREAD

**Issue by StewartJingga** [odpf/meteor#188]
**Is your feature request related to a problem? Please describe.**
When running long running extraction task, sometimes the issue happens on the sink part and those long extraction time is wasted.

**Describe the solution you'd like**
Enable validation of all plugins before running everything, especially processors and sink plugins

**Describe alternatives you've considered**
-

**Additional context**
-

**Comment by StewartJingga** [comment]
@ravisuhag @GrayFlash @scortier 

I created this issue a while ago, do you think we still need it?
since we move to stream and we have retry in place, this is barely a concern anymore.

Unless you want to have a dry run by extracting one data first and then run it.
But it would create complexity on the extractor since they would need to provide two functions, one to only extract one and another is for the main extraction.

Also when dry running, are we going to sink to the destination for real? or do we just send to console? if we just send to console then we are missing the whole point of the issue.

**Comment by ravisuhag** [comment]
We should still do init and connection checks on sinks and extractors by running init functions of all extractors, processors and sinks, right?

**Comment by StewartJingga** [comment]
Yes, right now it is being done in Init method on plugins. Unless, you want a more explicit lifecycle hook for a plugin.
